### ISSUE TITLE
 Need help on the steps to get a well trained py-faster-rCNN model

### ISSUE THREAD

**Issue by vj-1988** [rbgirshick/py-faster-rcnn#259]
I am trying to train faster-rcnn on my own dataset. My training steps are as follows: 

1) I took a pretrained imageNet VGG-16 caffemodel and fine-tuned it for my dataset (8 categories) for about 10K iterations. 

2) Then I ran py-Faster-rCNN on this fine-tuned model for about 80K iterations, by following the standard instructions

3) For training the faster-rCNN, specifically I am setting the following: 

a) average loss= 100 in solver.prototxt (final loss after 80k iterations was around 0.02)
b) changing the layer names from cls_score and bbox_pred to cls_score_bgs and bbox_pred_bgs
c) batchsize to 4 ( setting iter_size =4 in solver.prototxt)
d) setting dropout_ratio to 0 in train.prototxt

But I am facing these 2 problems in general: 
a) my bounding boxes are not accurate and tight. They are a little here and there around the region where the object is . Most of the times, the detected rectangular region covers larger area than the object region. 

b) when there are multiple objects in a scene to be detected, my model only detects 1 object accurately (>95%). When I set a lower confidence threshold (85%) in demo.py, the predictions are not right.

**Comment by kristellmarisse** [comment]
I was able to train my own dataset by following the instructions given in this page.

http://sunshineatnoon.github.io/Train-fast-rcnn-model-on-imagenet-without-matlab/

I did not change the layer names as you did, but still I am not able to get the object regions like the regions detected clearly in demo.py example. I trained my model for 120k iterations.

**Comment by manipopopo** [comment]
@vj-1988 

Have you changed the layer name `bbox_pred` in [`snapshot`](https://github.com/rbgirshick/py-faster-rcnn/blob/master/lib/fast_rcnn/train.py#L55) to `bbox_pred_bgs`? 

By default the training targets of bounding boxes (`bbox_pred`, or your `bbox_pred_bgs`) are [normalized](https://github.com/rbgirshick/py-faster-rcnn/blob/master/lib/rpn/proposal_target_layer.py#L140). If the normalization coefficients aren't saved into _caffemodel_, you will need to transform the output of `bbox_pred_bgs` manually during evaluation.

**Comment by vj-1988** [comment]
@manipopopo : I tried both the scenario. I changed the layer name bbox_pred and trained once and I trained another model without changing the bbox_pred name. In my case I got slightly better results for the model trained with out changing the layer name.

**Comment by jng075** [comment]
@vj-1988 how did you do the fine-tuning for the second point you mentioned in your first post? Could I email you to find out more? I have trained my custom dataset using petrained imagenet weights for 40,000 iterations. Now I want to fine-tune it further and improve mAP.

**Comment by steven20605** [comment]
@vj-1988  did u found a way to fix ur problem (a), i'm getting a low mAP because of that

**Comment by Ram-Godavarthi** [comment]
@vj-1988 
@user1103 
Even i have same question..
how did you do it??

Please let me know.

**Comment by Ram-Godavarthi** [comment]
HI Guys,
I am not able to view the output images with boxes.
I have trained the network using VGG pretrained model.
I have got the snapshot stored in output directory.
Now i am using it to check the output.. 
It is not displaying anything. But i am getting like below shown.. 




I0607 13:26:16.972672  3209 net.cpp:816] Ignoring source layer fc7_rad_drop7_0_split
I0607 13:26:16.972679  3209 net.cpp:816] Ignoring source layer cls_score_rad
I0607 13:26:16.972692  3209 net.cpp:816] Ignoring source layer bbox_pred_rad
I0607 13:26:16.972699  3209 net.cpp:816] Ignoring source layer loss_cls
I0607 13:26:16.972712  3209 net.cpp:816] Ignoring source layer loss_bbox


Loaded network /home/ubuntu/FRCN_ROOT/data/faster_rcnn_models/VGG16_cnn_m_1024_faster_rcnn_iter_5000.caffemodel
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/bicycle0007.jpg
Detection took 0.189s for 300 object proposals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/bicycle0012.jpg
Detection took 0.188s for 300 object proposals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/bicycle0018.jpg
Detection took 0.183s for 300 object proposals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/bicycle0025.jpg
Detection took 0.191s for 300 object proposals
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Demo for data/demo/bicycle0029.jpg
Detection took 0.184s for 300 object proposals


i am using AWS configured GPU instance with ubuntu 16.04..
What should i do to get the output displayed??
